tests/macros/src/main/scala/scala/macros/tests/sourcecode/SourceContext.scala

```diff
@@ -0,0 +1,77 @@
+package scala.macros.tests.sourcecode
+
+import scala.macros._
+import scala.language.implicitConversions
+
+// Unification of macro impl/defn forces us to put utilities in separate object
+private[sourcecode] object Util {
+  // TODO(olafur) replace with more typesafe typeOf[Line]
+  private def root: Term =
+    Term.Name("_root_").select("scala" :: "macros" :: "tests" :: "sourcecode" :: Nil)
+  def prefix(name: String): Term =
+    root.select(name :: Nil)
+
+  // Should this be handled by engines? I would claim that this is application specific
+  // for sourcecode.
+  def isSynthetic(s: Symbol)(implicit m: Mirror): Boolean = {
+    val name = s.name.value
+    name == "<init>" ||
+    name == "<root>" ||
+    (name.startsWith("<local ") && name.endsWith(">"))
+  }
```

Comment:
I think `isSynthetic` should be defined in macro system -- it's a query on Symbol, backend should implement that. In `Dotty`, synthetic symbols have a `Synthetic` flag.

Reply:
scalac also has `.isSynthetic`, but `<init>` an `<root>` are not synthetic according to sourcecode's use-case. I would claim this is application specific logic, and doesn't need to be handled by the macro engines.

Reply:
Ah I see, this is a macro-specific thing, thanks for explanation.